### PR TITLE
Fix Gradle 5.x deprecation warnings

### PR DESCRIPTION
**Pull request by tlusk**
This fixes issue #725

### CODE REVIEW

**Comment by cdancy** [comment]
I was actually going to do something similar and had at least one thing in mind I wanted to change. But since you're here.... ;)

Would you mind searching through our tasks to see where we do things like `project.objects.property(Map)` or something to that effect and replace with the new `project.objects.mapProperty(String, String)`?

**Comment by tlusk** [comment]
@cdancy yup, I just found that last warning during some testing (hence draft PR).  Just made the change to use `mapProperty` but it's failing some of the tests now so I'm investigating.

I'll update the PR with those changes once I get them working!

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy [hunk 1]
```diff
@@ -265,7 +266,7 @@ class DockerCreateContainer extends DockerExistingImage {
 
     void withEnvVar(def key, def value) {
         if (envVars.getOrNull()) {
-            envVars.get().put(key, value)
+            envVars.put(key, value)
```
Comment (tlusk) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy]: `MapProperty.get()` returns an `ImmutableMap` copy of the property.  Need to call `MapProperty.put()` directly in order to modify the property.

**Comment by tlusk** [comment]
@cdancy all set, finished testing and not seeing any more deprecation warnings.

**Comment by cdancy** [comment]
@tlusk just wondering: were there any test failures locally when running `./gradlew functionalTest`?

**Comment by tlusk** [comment]
@cdancy no failures, functional tests were all clean.

**Comment by cdancy** [comment]
@tlusk sounds good. CI hadn't kicked in here yet (just did now) and so was just wondering if anything ... _interesting_ ... popped.

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy [hunk 2]
```diff
@@ -29,6 +29,7 @@ import com.github.dockerjava.api.model.RestartPolicy
 import com.github.dockerjava.api.model.Volume
 import com.github.dockerjava.api.model.VolumesFrom
 import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.MapProperty
```
Comment (bmuschko) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy]: The use of `MapProperty` raises the minimum compatible Gradle version to 5.1. Please update the note in the user guide to reflect that.

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy [hunk 3]
```diff
@@ -110,7 +111,7 @@ class DockerCreateContainer extends DockerExistingImage {
 
     @Input
     @Optional
-    final Property<Map> envVars = project.objects.property(Map)
+    final MapProperty<Object, Object> envVars = project.objects.mapProperty(Object, Object)
```
Comment (bmuschko) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy]: This should define the concrete types. Let's use String/String here and in the code that passes the key/value pairs to Docker Java, we'll want to remove the support for Closures. There's no more need for Closures as the Property takes care of lazy evaluation.

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy [hunk 4]
```diff
@@ -153,7 +154,7 @@ class DockerCreateContainer extends DockerExistingImage {
 
     @Input
     @Optional
-    final Property<Map> binds = project.objects.property(Map)
+    final MapProperty<Object, Object> binds = project.objects.mapProperty(Object, Object)
```
Comment (bmuschko) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy]: Same comment as above in regard to typing.

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy [hunk 5]
```diff
@@ -204,7 +205,7 @@ class DockerCreateContainer extends DockerExistingImage {
 
     @Input
     @Optional
-    final Property<Map> labels = project.objects.property(Map)
+    final MapProperty<Object, Object> labels = project.objects.mapProperty(Object, Object)
```
Comment (bmuschko) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy]: Same comment as above in regard to typing.

**Comment by tlusk** [comment]
@bmuschko requested changes are complete.

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy (hunk 3 above)
Comment (tlusk) [reply src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy]: I did wonder about the map being untyped, but I originally decided to leave it alone in the event there was an intentional reason for it.

**Comment by cdancy** [comment]
Taking this in now. Thanks for the cleanup/fixes @tlusk !

**Pull request merged**